examples/mqtt-client/mqtt-client.c

```diff
@@ -591,12 +601,12 @@ connect_to_broker(void)
 static void
 ping_parent(void)
 {
-  if(uip_ds6_get_global(ADDR_PREFERRED) == NULL) {
-    return;
+  if(have_connectivity()) {
+    uip_icmp6_send(uip_ds6_defrt_choose(), ICMP6_ECHO_REQUEST, 0,
+                   ECHO_REQ_PAYLOAD_LEN);
+  } else {
+    /* we don't have connectivity; cannot send a ping */
```

Comment:
Do we need the else block here? (Unless we add some logging)

Reply:
Let me add `LOG_ERR` here :-)

Reply:
I wouldn't call this quite an error. More of a "not ready yet". I'd either warn or dbg.

Reply:
oh, alright!